widget/list.go

```diff
@@ -250,23 +251,52 @@ func (l *List) UnselectAll() {
 	}
 }
 
-func (l *List) visibleItemHeights(itemHeight float32, length int) (visible map[int]float32, offY float32, minRow, maxRow int) {
-	maxRow = length
+func (l *List) visibleItemHeights(itemHeight float32, length int) (visible []float32, offY float32, minRow int) {
+	maxRow := length
 	rowOffset := float32(0)
 	isVisible := false
-	visible = make(map[int]float32)
+	visible = []float32{}
 
 	if l.scroller.Size().Height <= 0 {
 		return
 	}
 
+	// theme.Padding is a slow call, so we cache it
+	padding := theme.Padding()
+
+	if len(l.itemHeights) == 0 {
+		paddedItemHeight := itemHeight + padding
+
+		offY = float32(math.Floor(float64(l.offsetY/paddedItemHeight))) * paddedItemHeight
+		minRow = int(math.Floor(float64(offY / paddedItemHeight)))
+		maxRow = int(math.Ceil(float64((offY + l.scroller.Size().Height) / paddedItemHeight)))
+
+		if minRow < 0 {
+			minRow = 0
+			offY = 0
+		}
+		if minRow > length {
```

Comment:
This doesn't seem right somehow. Why would the minimum row be the index 1 after the last item?

Reply:
Indeed.